### PR TITLE
Remove FromFrontend/ToCore

### PR DESCRIPTION
**Pull request by Acentelles**
Continuation of the work of making the code reflect the diagram below. Some of the work was discussed during EthCC with @mariari 
Screenshot 2021-07-26 at 11 23 09

- Remove ToCore inside the Translate module and create Pipeline/ToHR instead
- Separate temporary functions, datatypes, ... that live only inside a certain algorithm, such as in frontendToHR, to persisting code that belongs to a state in the pipeline. For example, Special def is only used in the algorithm from Sexp to HR and discarded afterwards, whereas the so called CoreDef is a RawGlobal that defines the HR state
- Remove `'` (ticks) from Core.Base types: This requires Andy's approval, since he's the creator of the library and this PR overwrites the default config value of `extensible`
- Add qualified imports
- Refactor Pipeline implementation such as typecheck
- Remove some re-exporting for consistency of imports. E.g. 
    - TC.Annotation (TC referring to CheckTerm)
    - IR.Annotation
    - Typed.Annotation (Typed referring to Typechecker)

### CODE REVIEW

**Review by mariari — APPROVED** [review]
A lot of code changes, overall it seems good. It's a good step forward for the compiler

**Review by cwgoes — COMMENTED** [review]
Concept ACK; should see review from @andy-morris

File: library/EasyPipeline/src/Easy.hs [hunk 1]
```diff
@@ -45,9 +45,10 @@ import qualified Juvix.Library.HashMap as Map
 import qualified Juvix.Library.NameSymbol as NameSymb
 import qualified Juvix.Pipeline as Pipeline
 import qualified Juvix.Pipeline.Compile as Compile
-import qualified Juvix.Pipeline.Core as Core
+import qualified Juvix.Pipeline.ToHR as ToHR
+import qualified Juvix.Pipeline.ToIR as ToIR
 import qualified Juvix.Sexp as Sexp
-import qualified Juvix.ToCore.Types as ToCore.Types
+-- import qualified Juvix.ToCore.Types as ToCore.Types
```
Comment (andy-morris) [inline library/EasyPipeline/src/Easy.hs:51]: can this just be removed?
```suggestion
```

File: library/Pipeline/src/Juvix/Pipeline/ToHR/Sig.hs [hunk 2]
```diff
@@ -1,35 +1,29 @@
-module Juvix.ToCore.FromFrontend.Transform.Sig (transformSig) where
+{-# LANGUAGE LiberalTypeSynonyms #-}
+{-# LANGUAGE RecordWildCards #-}
```
Comment (andy-morris) [inline library/Pipeline/src/Juvix/Pipeline/ToHR/Sig.hs:2]: can these be moved to the `package.yaml`? `UndecidableInstances` i was asked before to only enable explicitly in files that need it but these other two are probably fine

File: library/StandardLibrary/src/Juvix/Library/LineNum.hs [hunk 3]
```diff
@@ -1,3 +1,4 @@
+{-# LANGUAGE DeriveDataTypeable #-}
 {-# LANGUAGE FunctionalDependencies #-}
 {-# LANGUAGE TemplateHaskell #-}
```
Comment (andy-morris) [inline library/StandardLibrary/src/Juvix/Library/LineNum.hs:3]: can these be moved to `package.yaml` too?

**Review by andy-morris — APPROVED** [review]
looks good! a couple of tiny nitpicks but i selected approve anyway since what you want to do with those things is up to you

**Pull request merged**